### ISSUE TITLE
📎 Implement `lint/noGlobalEval` - `eslint/no-eval`

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#1088]
### Description

Implement [no-eval](https://eslint.org/docs/latest/rules/no-eval/) without rule options.

In contrast to the Eslint rule, I think we should not warn against local variables named `eval`.
In other words, teh following code should be valid:

```js
function f(eval) {
    eval("var a = 0");
};
```

However, we could report all use of `eval` including when we pass it to a function or assign it somewhere:

```js
f(eval) // should report this
const evalAlias = eval; // should report this
```

Want to contribute? Lets you know you are interested! We will assign you to the issue to prevent several people to work on the same issue. Don't worry, we can unassign you later if you are no longer interested in the issue! Read our [contributing guide](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md?rgh-link-date=2023-08-23T12%3A52%3A31Z) and [analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md?rgh-link-date=2023-08-23T12%3A52%3A31Z).

## Implementation notes

The implementor could use the `biome_js_syntax::global_identifier` helper function.

**Comment by you-5805** [comment]
@Conaclos Can I take this issue?
